lib/model/indexhandler.go

```diff
@@ -328,12 +348,42 @@ func (s *indexHandler) receive(fs []protocol.FileInfo, update bool, op string) e
 	if !update {
 		fset.Drop(deviceID)
 	}
+
+	l.Debugf("Received %d files for %s from %s, prevSeq=%d, lastSeq=%d", len(fs), s.folder, deviceID.Short(), prevSequence, lastSequence)
+
+	// Verify that the previous sequence number matches what we expected
+	if prevSequence > 0 && prevSequence != fset.Sequence(deviceID) {
+		l.Warnf("Bug: device %v folder %s sent index update with sequence %d while we expected %d", deviceID.Short(), s.folder, fset.Sequence(deviceID), s.prevSequence)
+		s.evLogger.Log(events.Failure, "index update with unexpected sequence")
```

Comment:
We could log with `ur.FailureReport` as data, and attach the mismatching sequences as extra data. Would be nice to see if there's any pattern (off by one, off by multiple of default patch size, ...), while still keeping the "uniform" event name for grouping in sentry.